#! /bin/sh /usr/share/dpatch/dpatch-run
## 37_msi_tv_anywhere.dpatch by Mario Limonciello <superm1@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add support for MSI TV@anywhere Master Remote (LP: #241830)

@DPATCH@
diff -urNad lirc~/doc/lirc.hwdb lirc/doc/lirc.hwdb
--- lirc~/doc/lirc.hwdb	2008-08-13 23:12:42.000000000 -0500
+++ lirc/doc/lirc.hwdb	2008-08-13 23:13:10.000000000 -0500
@@ -86,6 +86,7 @@
 KNC ONE TV Station (-/SE/PRO/RDS);knc_one;lirc_dev lirc_i2c;hw_default;knc_one/lircd.conf.knc_one;
 Lenco MXTV-9578 CP (card=50);pixelview_pak;devinput;hw_default;pixelview/lircd.conf.playtv_pro;
 Miro PCTV serial port receiver;pctv;none;hw_pinsys;pinnacle_systems/lircd.conf.pctv;
+MSI TV@nywhere Master;devinput;none;hw_default;msi/lircd.conf.tvanywhere;
 Nebula Digitv;leadtek;devinput;hw_default;nebula/digitv.conf;
 Phoebe Tv Master + FM (card=22);cph06x;devinput;hw_default;cph03x/lircd.conf.cph03x;
 Pinnacle Systems PCTV Sat receiver;pctv;none;hw_pinsys;pinnacle_systems/lircd.conf.pctv;
diff -urNad lirc~/remotes/msi/lircd.conf.tvanywhere lirc/remotes/msi/lircd.conf.tvanywhere
--- lirc~/remotes/msi/lircd.conf.tvanywhere	1969-12-31 18:00:00.000000000 -0600
+++ lirc/remotes/msi/lircd.conf.tvanywhere	2008-08-13 23:12:42.000000000 -0500
@@ -0,0 +1,86 @@
+# $Id: TVatnywhere,v 1.4 2006/04/19 05:50:12 grog Exp $
+#
+# Configuration file for MSI TV@nywhere Master using the tuner's IR
+# receiver.  This will probably also apply to the MSI TV@nywhere (not
+# "Master").  The current configuration file for the TV@nywhere remote
+# control interfaces via an external card.
+#
+# Contributed by Greg Lehey, 19 April 2006
+# brand:                       TV@nywhere Master
+# model no. of remote control:  not marked
+#
+# See http://www.lemis.com/grog/Thumbnails-20060419.html for a photo
+# of the remote control.
+#
+# To configure:
+#
+# 1.  Install a recent Linux kernel (I used 2.6.16) with the cx88
+#     driver.  This kernel should automatically detect card and IR
+#     control.
+#
+# 2.  The IR remote control will appear as a device in
+#     /dev/input/eventX.  To find out the value of X, check in
+#     /proc/bus/input/devices.  You should see something like:
+#
+#  I: Bus=0001 Vendor=1462 Product=8606 Version=0001
+#  N: Name="cx88 IR (MSI TV-@nywhere Master"
+#  P: Phys=pci-0000:01:08.0/ir0
+#  S: Sysfs=/class/input/input2
+#  H: Handlers=kbd event2
+#  B: EV=100003
+#  B: KEY=20c0000 800000 0 0 0 4 10000 100 0 1e0000 0 2 10100ffc
+#
+# The 2 in the line starting with H: gives the clue as to the device
+# name; it's /dev/input/event2.
+#
+# 3.  Put this file in /etc/lircd.conf (possibly along with others).
+#
+# 4.  Start lircd like this:
+#
+#     lircd -H dev/input -d /dev/input/event2
+
+begin remote
+
+  name  	TVatnywhere_Master
+  bits           16
+  eps            30
+  aeps          100
+
+  one             0     0
+  zero            0     0
+  pre_data_bits   16
+  pre_data       0x8001
+  gap          175846
+  toggle_bit      0
+
+
+      begin codes
+          FullScreen               0x0177
+          MTS                      0x00E2
+          Function                 0x0021
+          Power                    0x0074
+          1                        0x0002
+          2                        0x0003
+          3                        0x0004
+          4                        0x0005
+          5                        0x0006
+          6                        0x0007
+          7                        0x0008
+          8                        0x0009
+          9                        0x000A
+          0                        0x000B
+          ChUp                     0x0192
+          ChDown                   0x0193
+          VolUp                    0x0073
+          VolDown                  0x0072
+          T-shift                  0x0014
+          Return                   0x001C
+          Rewind                   0x00A8
+          Play                     0x0199
+          FFWD                     0x00D0
+          Mute                     0x0071
+      end codes
+
+end remote
+
+
